deepfence_server/reporters/scan/scan_result_actions.go

```diff
@@ -223,6 +223,39 @@ func DeleteScan(ctx context.Context, scanType utils.Neo4jScanType, scanId string
 	return nil
 }
 
+func StopCloudComplianceScan(ctx context.Context, scanType, scanId string) error {
+
+	driver, err := directory.Neo4jClient(ctx)
+	if err != nil {
+		return err
+	}
+	session := driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeWrite})
+	defer session.Close()
+
+	tx, err := session.BeginTransaction(neo4j.WithTxTimeout(15 * time.Second))
+	if err != nil {
+		return err
+	}
+	defer tx.Close()
+
+	query := `MATCH (n:%s) -[:SCANNED]-> ()
+        WHERE n.node_id = $scan_id
+        AND n.status = $in_progress
+        SET n.status=$cancel_pending, n.stop_requested=true`
```

Comment:
You can see here how the `stop_requested` is redundant with the `cancel_pending` status

Reply:
Actually this is not going to work :( 
Reason being, an already running scan will continuously emit "IN PROGRESS" asynchronously
Hence there's a chance we might overwrite the "CANCEL_PENDING" status with IN PROGRESS" even before we check this scan. 
This will result in the stop never getting triggered. 
Hence we would need this flag for deterministic behavior.